paddle/fluid/framework/ir/mkldnn/mkldnn_pass_util.h

```diff
@@ -77,6 +107,58 @@ static void GetInfoFromTheFirstOp(
   }
 }
 
+static void GetInfoFromTheFirstOp(ir::Graph* graph,
+                                  const std::string& flag,
+                                  const std::string& key_suffix,
+                                  StringPairMap* info_map) {
+  VLOG(3) << "get variables from the first op's attr";
+  const std::string unsigned_flag = "_unsigned";
+  const std::string suffix = "_" + key_suffix + "_" + flag;
+  const std::string suffix_is_unsigned = suffix + unsigned_flag;
+  for (auto* op_node :
+       ir::TopologyVarientSort(*graph, static_cast<ir::SortKind>(0))) {
+    if (!op_node->IsOp() || op_node->Op()->Type() == "feed" ||
+        op_node->Op()->Type() == "fetch")
+      continue;
+
+    auto* op_desc = op_node->Op();
+    if (op_desc->GetAttrIfExists<bool>(flag)) {
+      op_desc->RemoveAttr(flag);
+      std::vector<std::string> attr_names = op_desc->AttrNames();
+      for (auto fake_name : attr_names) {
+        auto is_unsigned = false;
+        size_t pos = fake_name.find(suffix_is_unsigned);
+
+        if (pos != std::string::npos) {
+          std::string unsigned_var_name = fake_name;
+          is_unsigned =
+              PADDLE_GET_CONST(bool, op_desc->GetAttr(unsigned_var_name));
+
+          std::string var_name = fake_name.substr(0, pos);
+          size_t unsigned_pos = fake_name.find(unsigned_flag);
+          std::string vector_name =
+              fake_name.erase(unsigned_pos, unsigned_flag.length());
+          auto scales_vector = PADDLE_GET_CONST(std::vector<float>,
+                                                op_desc->GetAttr(vector_name));
+
+          LoDTensor tensor;
+          const int size = static_cast<int>(scales_vector.size());
+          auto* data =
+              tensor.mutable_data<double>({size}, platform::CPUPlace());
+          for (int i = 0; i < size; i++) {
+            data[i] = static_cast<double>(scales_vector[i]);
+          }
```

Comment:
Consider range constructor like in here https://stackoverflow.com/a/6399098

Reply:
And Why do we cast float to double?

Reply:
Sometimes we converted to double when we did a few scale operations, but here it does not apply, I will change to float as it was